services/graph.js

```diff
@@ -1,136 +1,227 @@
-      .then((res) => {
-        if (res.ok) return res.json();
-        throw new Error('Unable to fetch the most viewed datasets');
-      })
-      .then(res => res.data.map(d => d.dataset));
-  }
-
-  /**
-   * Get the list of most favourited datasets
-   * @returns {Promise<string[]>} List of sorted ids
-   */
-  getMostFavoritedDatasets() {
-    return fetch(
-      `${process.env.WRI_API_URL}/graph/query/most-liked-datasets?application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`,
-      { headers: { 'Upgrade-Insecure-Requests': 1 } }
-
-    )
-      .then((res) => {
-        if (res.ok) return res.json();
-        throw new Error('Unable to fetch the most favourited datasets');
-      })
-      .then(res => res.data.map(d => d.id));
-  }
-}
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error in count dataset view ${datasetId}: ${status}: ${statusText}`);
+      throw new Error(`Error in count dataset view ${datasetId}: ${status}: ${statusText}`);
+    });
+};
+
+/**
+ * Get the list of most viewed datasets
+ * @returns {Promise<string[]>} List of sorted ids
+ */
+export const fetchMostViewedDatasets = (params = {
+  env: process.env.API_ENV,
+  application: process.env.APPLICATIONS
+}) => {
+  logger.info('Fetch most viewed datasets');
+  return WRIAPI.get('graph/query/most-viewed',
+    {
+      params,
+      headers: { 'Upgrade-Insecure-Requests': 1 }
+    })
+    .then(response => WRISerializer(response.data))
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching most viewed datasets: ${status}: ${statusText}`);
+      throw new Error(`Error fetching most viewed datasets: ${status}: ${statusText}`);
+    });
+};
+
+/**
+ * Get the list of most favourited datasets
+ * @returns {Promise<string[]>} List of sorted ids
+ */
+export const fetchMostFavoritedDatasets = (params = {
+  env: process.env.API_ENV,
+  application: process.env.APPLICATIONS
+}) => {
+  logger.info('Fetch most favorited datasets');
+  return WRIAPI.get('graph/query/most-liked-datasets',
+    {
+      params,
+      headers: { 'Upgrade-Insecure-Requests': 1 }
+    })
+    .then(response => WRISerializer(response.data))
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching most favorited datasets: ${status}: ${statusText}`);
+      throw new Error(`Error fetching most favorited datasets: ${status}: ${statusText}`);
+    });
+};
+
+export const fetchSimilarDatasets = (datasetIds, withAncestors = true, params = {
+  published: true,
+  env: process.env.API_ENV,
+  application: process.env.APPLICATIONS,
+  limit: 6
+}) => {
+  logger.info('Fetch similar datasets');
+  const endpoint = withAncestors ? 'similar-dataset-including-descendent' : 'similar-dataset';
+  return WRIAPI.get(
+    `graph/query/${endpoint}`,
+    {
+      params: {
+        ...params,
+        dataset: datasetIds.join(',')
```

Comment:
Should be part of the `params` object. The fewer operation the service has to perform, the better.